### PR TITLE
[AssemblyProcessor] - Remove unused options and deadcode

### PR DESCRIPTION
**Pull request by Jklawreszuk**
# PR Details
## Description

As the title suggests.

## Motivation and Context

Simplification of target/props files to get a clearer picture of building process.

## Types of changes

- [x] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My change requires a change to the documentation.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] **I have built and run the editor to try this change out.**

### CODE REVIEW

File: sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj [hunk 1]
```diff
@@ -14,7 +14,6 @@
     <UseVSHostingProcess>false</UseVSHostingProcess>
     <DocumentationFile>
     </DocumentationFile>
-    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (Ethereal77) [inline sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj:17]: This `<TreatWarningsAsErrors>` is not an AssemblyProcessor option, but a C# compiler option.

File: sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj [hunk 2]
```diff
@@ -37,7 +36,6 @@
     <AllowUnsafeBlocks>true</AllowUnsafeBlocks>
     <DocumentationFile>
     </DocumentationFile>
-    <TreatWarningsAsErrors>false</TreatWarningsAsErrors>
```
Comment (Ethereal77) [inline sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj:40]: Same as above.

File: sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj [hunk 3]
```diff
@@ -61,7 +59,6 @@
...
```
Comment (Ethereal77) [inline sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj:64]: Same as above.

File: sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj (hunk 1 above)
Comment (Jklawreszuk) [reply sources/tools/Stride.TextureConverter.Wrappers/FreeImage.Net/FreeImage.NET.csproj]: My mistake, Changed now 😅

File: sources/core/Stride.Core/build/Stride.Core.targets [hunk 4]
```diff
@@ -72,15 +72,15 @@
       <StrideAddReference Include="@(ReferencePath)" Condition="'%(ReferencePath.NuGetPackageId)' == ''"/>
     </ItemGroup>
     <PropertyGroup>
-      <StrideAssemblyProcessorOptions Condition="'$(StrideAssemblyProcessorOptions)' == ''">--auto-notify-property --parameter-key --auto-module-initializer --serialization</StrideAssemblyProcessorOptions>
+      <StrideAssemblyProcessorOptions Condition="'$(StrideAssemblyProcessorOptions)' == ''"> --parameter-key --auto-module-initializer --serialization</StrideAssemblyProcessorOptions>
```
Comment (Ethereal77) [inline sources/core/Stride.Core/build/Stride.Core.targets:75]: Nothing important, but the space before `--parameter-key` 😁

File: sources/targets/Stride.Core.props [hunk 5]
```diff
@@ -127,7 +127,7 @@
   <!-- Default values -->
   <PropertyGroup>
     <StrideAssemblyProcessor Condition="'$(StrideAssemblyProcessor)' == ''">false</StrideAssemblyProcessor>
-    <StrideAssemblyProcessorOptions Condition="'$(StrideAssemblyProcessorOptions)' == ''">--auto-notify-property --auto-module-initializer --serialization</StrideAssemblyProcessorOptions>
+    <StrideAssemblyProcessorOptions Condition="'$(StrideAssemblyProcessorOptions)' == ''"> --auto-module-initializer --serialization</StrideAssemblyProcessorOptions>
```
Comment (Ethereal77) [inline sources/targets/Stride.Core.props:130]: Same, the space

File: sources/targets/Stride.props [hunk 6]
```diff
@@ -3,7 +3,7 @@
   <PropertyGroup>
     <StrideSdkTargets>$(MSBuildThisFileDirectory)Stride.targets</StrideSdkTargets>
     <StrideAssemblyProcessorGlobal>true</StrideAssemblyProcessorGlobal>
-    <StrideAssemblyProcessorDefaultOptions>--auto-notify-property --parameter-key --auto-module-initializer --serialization</StrideAssemblyProcessorDefaultOptions>
+    <StrideAssemblyProcessorDefaultOptions> --parameter-key --auto-module-initializer --serialization</StrideAssemblyProcessorDefaultOptions>
```
Comment (Ethereal77) [inline sources/targets/Stride.props:6]: Same, the space.

**Review by Ethereal77 — APPROVED** [review]
Appart from the space nitpick, LGTM.
Good job.

File: sources/core/Stride.Core/build/Stride.Core.targets (hunk 4 above)
Comment (Jklawreszuk) [reply sources/core/Stride.Core/build/Stride.Core.targets]: Here you go 😁

**Comment by Jklawreszuk** [comment]
Could this PR please be merged please?

**Review by Kryptos-FR — APPROVED** [review]

**Comment by Jklawreszuk** [comment]
@Kryptos-FR Glad to hear 😅 So It can be merged now right?

**Pull request merged**